extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -29,35 +64,47 @@ export class WorkspaceService implements IWorkspaceService {
 	}
 
 	async addProjectsToWorkspace(projectFiles: vscode.Uri[]): Promise<void> {
-		if (vscode.workspace.workspaceFile) {
-			const currentProjects: vscode.Uri[] = this.getProjectsInWorkspace();
-			const newWorkspaceFolders: string[] = [];
-			let newProjectFileAdded = false;
-			for (const projectFile of projectFiles) {
-				if (currentProjects.findIndex((p: vscode.Uri) => p.fsPath === projectFile.fsPath) === -1) {
-					currentProjects.push(projectFile);
-					newProjectFileAdded = true;
-
-					// if the relativePath and the original path is the same, that means the project file is not under
-					// any workspace folders, we should add the parent folder of the project file to the workspace
-					const relativePath = vscode.workspace.asRelativePath(projectFile, false);
-					if (vscode.Uri.file(relativePath).fsPath === projectFile.fsPath) {
-						newWorkspaceFolders.push(path.dirname(projectFile.path));
-					}
-				}
-			}
+		if (!projectFiles || projectFiles.length === 0) {
+			return;
+		}
 
-			if (newProjectFileAdded) {
-				// Save the new set of projects to the workspace configuration.
-				await this.setWorkspaceConfigurationValue(ProjectsConfigurationName, currentProjects.map(project => this.toRelativePath(project)));
-				this._onDidWorkspaceProjectsChange.fire();
+		// a workspace needs to be open to add projects
+		if (!vscode.workspace.workspaceFile) {
```

Comment:
hmm, the warning prompt should be added when we launch the create/open project dialog, right? if we do this here, for create project scenario, there will be orphaned files created.

Reply:
that's a good point. I'll check with Raden on if the warning should be a toast notification that shows when the dialogs are opened or if it should be warning text in the dialog

Reply:
I moved the warning text to be shown when after the "OK" button is clicked on the dialogs and before the project gets created so that there won't be the orphaned files problem. Raden said he'll think more about if this warning text should be a part of the dialog.

Reply:
My suggestion is to show the warning either before opening the dialogs or show it within the dialog, it would be frustrating if the user has already finished providing information but has to choose 'No' (e.g. to save some editor windows)

Reply:
Yeah that's understandable, but in the meanwhile, Raden thinks it's fine to have the toast notification after the clicking Ok as opposed to having it show when the dialog is opened. He'll think more about if the warning text should also be in the dialog itself. Eventually, we would want something that blocks the dialog like this (but we don't currently have this component): 
![image](https://user-images.githubusercontent.com/31145923/96517042-0b6d5c00-121d-11eb-96d8-83982bee405b.png)